### PR TITLE
Solar Devils ERT [PvE Marines]

### PR DESCRIPTION
**Pull request by Steelpoint**
# About the pull request

In marking the merger of the PvP and PvE servers under a single banner, this PR sets to celebrate that by the addition of the 'Solar Devils' battalion as a (currently admin only) Emergency Response Team. The Solar Devils are similarly well equipped and trained as they appear in PvE.

This PR also adds some content from the PvE server to enhance the appearance of the Solar Devils to better fit their appearance, such as a battalion patch sprite. 

These squads will spawn with, depending on if they are a half or full squad, 4 Riflemen, 2 Smartgunners, 2 Team Leaders, 1 Platoon Corpsman and 1 Platoon Leader. The Platoon Leader will only spawn with the 'Full Squad' ERT. 

The squad are all armed with M41A Mk1's.

----

In terms of function. This PR adds two distress calls for admins to use, a Half-Squad and Full Squad deployment of either 5 or 10, respectively, Solar Devil Marines. They spawn with some unique equipment, and they all possess increased skills. 

# Explain why it's good for the game

Show of comradeship, and a small easter egg of the PvE server in PvP. 

# Testing Photographs and Procedure
Screenshot 2024-07-28 21 52 07

# Changelog
:cl:
add: In observance of the merger of the PvP and PvE servers of CM-SS13. The 'Solar Devils', the USCM battalion that appear in the PvE server, have been added as an admin only Emergency Response Team. 
add: The Solar Devils are highly trained and well equipped by Marine standards, they will also spawn as part of their own squad. 
imageadd: A 'Solar Devils' uniform patch accessory has been added. 
/:cl:

### CODE REVIEW

**Comment by sunofang** [comment]
We're so back bros.

But in all seriousness, why is the distress call admin-only?

**Comment by Steelpoint** [comment]
> We're so back, PVE bros.
> 
> But in all seriousness, why is the distress call admin-only?

Its easier to have something added if it does not affect the core-gameplay loop, furthermore this will not dilute the ERT pool further. If the power's that be judge this worthy of being in the main hijack loop I'll accede, even if I have to nerf their equipment.

**Comment by Merrgear** [comment]
Hi, gods gift to pve, literally the coolest gm ever here, you didnt give them spec buck and instead gave them the RA no one uses, marked for death

In all seriousness tho i think this is a cool little reference admins can throw, but id not be too in support of standardizing it for a couple reasons
1, It kind of adds to the illusion that the Sun Riders on the GArrow are better than the avg marine if they are in here rocking the improved skills and spawned on a regular basis, as if they were added to normal ERT. PVP refugees already struggle a bit with this cause they are used to the less lore accurate skillset
2, The RA being in gunny's hands, ESPECIALLY only him being trained to use it is something we have been trying to fight since it was introduced, it is not the gunny weapon, gunny is not for combat, gunny is supposed to issue it out, would make sense moreso as in a random rfm's hands than gunny. also wall piercing ammo would be much more accurate than the incind and normal mags, since thats the RA's gimmick (along with a stun but that wont transfer without code)
3, Smartgunners in pvp are so wimpy compared to PVE and rock different IFF, it would serve to further confuse players who come from pvp and go "well the ERT has it like that!!!!" 

and the worst sin of all......its a platoon not a squad!!!!!!!!!!!!!!!!! marked for death in pve discord await your judgement

**Comment by Steelpoint** [comment]
@Merrgear 

It is worth taking into consideration I've somewhat moulded the squad within a balance framework of the PvP environment it is going to find itself playing in. 

Whilst I appreciate that there are far more differences to a PvE versus PvP marine that this PR doesn't really touch, but I have neither the skills nor the time to make those differences far more pronounced than what is in the PR. 

Nonetheless I will take your feedback in consideration and re-theme it that the Platoon Squad Leader won't spawn with a custom M4RA. At this stage I'll simply make all the Marines spawn with a M41A Mk1. I can not really buff the Smartgunner since that would require a lot of code work to implement the PvE SG mechanics for 1 role, same goes for the sentry guns the TL's spawn with.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

File: code/modules/gear_presets/uscm.dm [hunk 1]
```diff
@@ -994,3 +994,332 @@
...
+		new_human.equip_to_slot_or_del(new new_mask, WEAR_FACE)
+
+	var/list/possible_glasses = list(/obj/item/clothing/glasses/regular, /obj/item/clothing/glasses/regular/hipster, /obj/item/clothing/glasses/sunglasses, /obj/item/clothing/glasses/sunglasses/aviator, /obj/item/clothing/glasses/sunglasses/big) + subtypesof(/obj/item/clothing/glasses/mgoggles)
+	if(prob(50))
+		var/obj/item/clothing/mask/new_glasses = pick(possible_glasses)
+		new_human.equip_to_slot_or_del(new new_glasses, WEAR_EYES)
+
+/datum/equipment_preset/uscm/proc/spawn_marine_armor(mob/living/carbon/human/new_human)
+	if(prob(66))
+		new_human.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/marine/medium, WEAR_JACKET)
```
Comment (Diegoflores31) [inline code/modules/gear_presets/uscm.dm:1018]: give them veteran armor heheh ( its the same . just different name)

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/gear_presets/uscm.dm (hunk 1 above)
Comment (Steelpoint) [reply code/modules/gear_presets/uscm.dm:986]: Sadly vet armour does not match regular camo.

**Comment by Steelpoint** [comment]
Denoted changes made to better bring lore in-line with PvE.

**Review by Blundir — CHANGES_REQUESTED** [review]
find a replacement for the patch sprite, current one looks messy

**Comment by Steelpoint** [comment]
Replacement patch sprite added, thanks to @Blundir

**Review by Blundir — APPROVED** [review]
sprites look good

**Comment by Steelpoint** [comment]
I'd be interested in allowing a half-squad to spawn as a ERT distress call but I'll leave the PR as is, unless the power's that be decide otherwise.

File: code/datums/skills/uscm.dm [hunk 2]
```diff
@@ -426,3 +426,82 @@ COMMAND STAFF
...
+		SKILL_POWERLOADER = SKILL_POWERLOADER_TRAINED,
+		SKILL_VEHICLE = SKILL_VEHICLE_SMALL,
+		SKILL_JTAC = SKILL_JTAC_BEGINNER,
+		SKILL_ENDURANCE = SKILL_ENDURANCE_TRAINED,
+		SKILL_MEDICAL = SKILL_MEDICAL_MEDIC,
+		SKILL_SURGERY = SKILL_SURGERY_NOVICE,
+	)
+
+/datum/skills/smartgunner_PvE
+	name = "Fireteam Leader"
```
Comment (private-tristan) [inline code/datums/skills/uscm.dm:463]:
```suggestion
	name = "Smartgunner"
```
last time I checked smartgunners are smartgunners.

File: code/datums/skills/uscm.dm [hunk 3]
```diff
@@ -426,3 +426,82 @@ COMMAND STAFF
...
+		SKILL_CONSTRUCTION = SKILL_CONSTRUCTION_ENGI,
+		SKILL_ENGINEER = SKILL_ENGINEER_ENGI,
+		SKILL_LEADERSHIP = SKILL_LEAD_EXPERT,
+		SKILL_MEDICAL = SKILL_MEDICAL_TRAINED,
+		SKILL_SURGERY = SKILL_SURGERY_NOVICE,
+		SKILL_VEHICLE = SKILL_VEHICLE_SMALL,
+		SKILL_JTAC = SKILL_JTAC_TRAINED,
+		SKILL_INTEL = SKILL_INTEL_TRAINED,
+		SKILL_PILOT = SKILL_PILOT_EXPERT,
+	)
```
Comment (private-tristan) [inline code/datums/skills/uscm.dm:507]:
```suggestion
	name = "Squad Leader"
	skills = list(
		SKILL_ENGINEER = SKILL_ENGINEER_ENGI,
		SKILL_CONSTRUCTION = SKILL_CONSTRUCTION_ENGI,
		SKILL_FIREMAN = SKILL_FIREMAN_EXPERT,
		SKILL_POWERLOADER = SKILL_POWERLOADER_TRAINED,
		SKILL_ENDURANCE = SKILL_ENDURANCE_MASTER,
		SKILL_CQC = SKILL_CQC_TRAINED,
		SKILL_LEADERSHIP = SKILL_LEAD_EXPERT,
		SKILL_MEDICAL = SKILL_MEDICAL_TRAINED,
		SKILL_SURGERY = SKILL_SURGERY_NOVICE,
		SKILL_VEHICLE = SKILL_VEHICLE_SMALL,
		SKILL_JTAC = SKILL_JTAC_TRAINED,
		SKILL_INTEL = SKILL_INTEL_TRAINED,
		SKILL_PILOT = SKILL_PILOT_EXPERT,
	)
```
probably shouldn't have the same skills twice?

File: code/datums/skills/uscm.dm [hunk 4]
```diff
@@ -426,3 +426,80 @@ COMMAND STAFF
...
 		SKILL_INTEL = SKILL_INTEL_TRAINED,
 	)
+
+/*
+---------------------
+SOLAR DEVILS (PvE Marines)
+---------------------
+*/
+
+/datum/skills/rifleman_PvE
```
Comment (Zonespace27) [inline code/datums/skills/uscm.dm:436]: typepaths should be lowercase (goes for all of em)

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by Steelpoint** [comment]
Issues attended to.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/gear_presets/uscm.dm [hunk 5]
```diff
@@ -962,3 +962,326 @@
...
+		new_human.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/marine/heavy, WEAR_JACKET)
+
+/datum/equipment_preset/uscm/proc/spawn_marine_backpack(mob/living/carbon/human/new_human)
+	if(prob(75))
+		new_human.equip_to_slot_or_del(new /obj/item/storage/backpack/marine/satchel, WEAR_BACK)
+	else
+		new_human.equip_to_slot_or_del(new /obj/item/storage/backpack/marine, WEAR_BACK)
+
+/datum/equipment_preset/uscm/proc/spawn_marine_sidearm(mob/living/carbon/human/new_human)
+	return
```
Comment (harryob) [inline code/modules/gear_presets/uscm.dm:999]: why does this exist?

File: code/modules/gear_presets/uscm.dm [hunk 6]
```diff
@@ -962,3 +962,326 @@
...
+
+/datum/equipment_preset/uscm/proc/spawn_marine_armor(mob/living/carbon/human/new_human)
+	if(prob(66))
+		new_human.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/marine/medium, WEAR_JACKET)
+	else if(prob(50))
+		new_human.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/marine/light, WEAR_JACKET)
+	else
+		new_human.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/marine/heavy, WEAR_JACKET)
+
+/datum/equipment_preset/uscm/proc/spawn_marine_backpack(mob/living/carbon/human/new_human)
```
Comment (harryob) [inline code/modules/gear_presets/uscm.dm:992]: when is this called?

File: code/modules/gear_presets/uscm.dm [hunk 7]
```diff
@@ -962,3 +962,326 @@
...
+	if(prob(50))
+		var/obj/item/clothing/mask/new_mask = pick(possible_masks)
+		new_human.equip_to_slot_or_del(new new_mask, WEAR_FACE)
+
+	var/list/possible_glasses = list(/obj/item/clothing/glasses/regular, /obj/item/clothing/glasses/regular/hipster, /obj/item/clothing/glasses/sunglasses, /obj/item/clothing/glasses/sunglasses/aviator, /obj/item/clothing/glasses/sunglasses/big) + subtypesof(/obj/item/clothing/glasses/mgoggles)
+	if(prob(50))
+		var/obj/item/clothing/mask/new_glasses = pick(possible_glasses)
+		new_human.equip_to_slot_or_del(new new_glasses, WEAR_EYES)
+
+/datum/equipment_preset/uscm/proc/spawn_marine_armor(mob/living/carbon/human/new_human)
```
Comment (harryob) [inline code/modules/gear_presets/uscm.dm:984]: when is this called?

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Steelpoint** [comment]
Removed leftover concepts.

**Review by harryob — APPROVED** [review]

**Pull request merged**